### DISCUSSION TITLE
eepromWrapper doesn't seem to save to eeprom

### DISCUSSION THREAD

**Discussion by emaayan** [davetcc/tcMenu#357] [Q&A]
i have the following code:
it is initlized with     ArduinoEEPROMAbstraction eepromWrapper = ArduinoEEPROMAbstraction(&EEPROM);

  ```
eepromWrapper.write32(romStart + offset, value);    
   EEPROM.commit();
   uint32_t  t= eepromWrapper.read32(romStart + offset);
   Serial.print(t);
```

but t always seems to be zero, (this is an esp32) am i missing something? )

**Comment by davetcc** [comment]
Please try https://github.com/davetcc/IoAbstraction/blob/master/examples/arduinoEEPROMExample/arduinoEEPROMExample.ino which I tested as working this morning, pay close attention to the begin operation at the start of setup, ESP rom operations do not work without it.

**Reply by emaayan** [reply]
hmm i i'm betting an error on the begin
Unable to open NVS namespace  with error number 4353 must be an esp32 issue .

**Reply by davetcc** [reply]
How many times have you written to that EEPROM, like I said the other day there are a limited number of writes to it, a few thousand.

**Reply by davetcc** [reply]
If you want to write to it more often why not use a cheap as chips i2c AT24x device, the are fully supported and support orders of magnitude more writes.

**Reply by emaayan** [reply]
actualy till now i don't i've actually written anything to it at all. i was mainly testing classes at all, the objective is still to write rarely to it. 
because of the issues with end edit and onCommit i am calling save on the fnInvoke, but that does't calll saveEEProm directly but it first save the value into a member variable , and only if there's a difference between the member variable and the incoming paramter, the save eepRom will be invoke, so fn_invoke can be called a million times but the actuall save will only happen once. 
but at any rate, i don't think i've managed untill now to actually call save anywaym because i've never called commit 

i've tried easing flash, and now i can't seem to actually upload anything , (i was using  esp idf debugger) i'm also using littleFS to upload a bunch a json files to it. so i've configured the board_build.filesystem = littlefs 
wondering if erasing flash didn't some damage to it.
